### ISSUE TITLE
Connection: close header can cause truncated messages

### ISSUE THREAD

**Issue by JakePi3** [ztellman/aleph#181]
When sent a GET request with a `Connection: close` header, only the first 256KB are transferred and the rest of the data is silently dropped.

Data is being properly extracted, content-length is correctly set on response, but packet inspection reveals incomplete data sent.  Tested from multiple clients.

Can't pin this on Aleph, it could be an underlying Netty problem.

**Comment by ztellman** [comment]
Which version are you using?

**Comment by JakePi3** [comment]
0.4.0

**Comment by JakePi3** [comment]
Also, it may be relevant, I was sending a byte array in the `:body` of my response, generated using `byte-streams/to-byte-array`

**Comment by ztellman** [comment]
Can you target `0.4.1-beta1` and tell me if that solves the issue for you?
In 0.4.0 I was assuming the channel being "writable" would mean any write
would complete immediately, which isn't true for sufficiently large chunks
of data.  This is fixed in `0.4.1`.

**Comment by JakePi3** [comment]
Just verifying the issue with Alpeh 0.4.0:

```
C:\>curl -v -H "Connection: close" -G http://localhost:3000/data/raven/large > Temp\test
* STATE: INIT => CONNECT handle 0x80071bd8; line 1034 (connection #-5000)
* Added connection 0. The cache now contains 1 members
*   Trying 127.0.0.1...
* STATE: CONNECT => WAITCONNECT handle 0x80071bd8; line 1087 (connection #0)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to localhost (127.0.0.1) por
3000 (#0)
* STATE: WAITCONNECT => DO handle 0x80071bd8; line 1229 (connection #0)
> GET /data/raven/large HTTP/1.1
> User-Agent: curl/7.41.0
> Host: localhost:3000
> Accept: */*
> Connection: close
>
* STATE: DO => DO_DONE handle 0x80071bd8; line 1314 (connection #0)
* STATE: DO_DONE => WAITPERFORM handle 0x80071bd8; line 1441 (connection #0)
* STATE: WAITPERFORM => PERFORM handle 0x80071bd8; line 1454 (connection #0)
* HTTP 1.1 or later with persistent connection, pipelining supported
< HTTP/1.1 200 OK
< Content-Type: application/octet-stream
* Server Aleph/0.4.0 is not blacklisted
< Server: Aleph/0.4.0
< Connection: Close
< Date: Wed, 28 Oct 2015 10:06:39 GMT
< content-length: 1927956
<
{ [16223 bytes data]
* nread <= 0, server closed connection, bailing
* transfer closed with 1534904 bytes remaining to read
* Closing connection 0
* The cache now contains 0 members
* Expire cleared
curl: (18) transfer closed with 1534904 bytes remaining to read

C:\>dir Temp\test
 Volume in drive C is System
 Volume Serial Number is 02D0-3ABF

 Directory of C:\Temp

28/10/2015  10:06           393,052 test
               1 File(s)        393,052 bytes
               0 Dir(s)   2,606,338,048 bytes free
```

Going to test against `0.4.1-beta1`

**Comment by JakePi3** [comment]
Could not reproduce in 0.4.1-beta1 (same code base) so it appears to be fixed.

Just an aside, the server is still reporting its version as 0.4.0 in the response:

```
< Server: Aleph/0.4.0
```

You may consider extracting the version info via `(io/resource "/META-INF/maven/aleph/aleph/pom.properties)` or similar.
